analysis/org.eclipse.tracecompass.analysis.chromium.core/src/org/eclipse/tracecompass/internal/analysis/chromium/core/callstack/ChromiumCallStackProvider.java

```diff
@@ -92,20 +92,30 @@ protected boolean considerEvent(@NonNull ITmfEvent event) {
 
     @Override
     protected @Nullable ITmfStateValue functionEntry(@NonNull ITmfEvent event) {
-        if (event instanceof TraceEventEvent && 'B' == (((TraceEventEvent) event).getField().getPhase())) {
+        if (event instanceof TraceEventEvent && isEntry(event)) {
             return TmfStateValue.newValueString(event.getName());
         }
         return null;
     }
 
+    private static boolean isEntry(ITmfEvent event) {
+        char phase = ((TraceEventEvent) event).getField().getPhase();
```

Comment:
Layouts, yes, in the master version. This is a reference implementation to prove that it is possible using trace-event events.